### PR TITLE
[Backup] Migrate backup to track2 SDK

### PR DESCRIPTION
**Pull request by PARADISSEEKR**
**Description**

Here are main changes for Backup migration to track2:
- In track2, use HttpResponseError instead of CloudError.
- Modify `test_afs_xxx` to live test.
- Update `test_wl_sql_xxx` ,delete some operations that cannot be completed by live test(register,unregister).

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Backup

**Comment by yungezz** [comment]
hi @PARADISSEEKR what's status of the PR?

**Comment by yungezz** [comment]
cc @qwordy

**Comment by qwordy** [comment]
Some test failed.
https://clitestresultstac.blob.core.windows.net/20210521-055502-506086/backup.report.parallel.html

**Comment by qwordy** [comment]
I ran it again.
https://clitestresultstac.blob.core.windows.net/20210521-122625-720256/backup.report.parallel.html
@PARADISSEEKR Could you verify the correctness of the two failed test cases? Thanks.

**Review by qwordy — APPROVED** [review]

File: src/azure-cli/requirements.py3.windows.txt [hunk 1]
```diff
@@ -1,6 +1,6 @@
 adal==1.2.7
 antlr4-python3-runtime==4.7.2
-applicationinsights==0.11.7
+applicationinsights==0.11.9
```
Comment (qwordy) [inline src/azure-cli/requirements.py3.windows.txt:3]: Why modify version other than recoveryservices?

Comment (PARADISSEEKR) [reply src/azure-cli/requirements.py3.windows.txt:3]: There is an unknown problem in the virtual environment, then change it back

File: src/azure-cli/azure/cli/command_modules/backup/tests/latest/preparers.py [hunk 2]
```diff
@@ -281,7 +281,7 @@ def _get_resource_group(self, **kwargs):
         try:
             return kwargs.get(self.resource_group_parameter_name)
         except KeyError:
-            template = 'To create an item, a resource group is required. Please add ' \
+            template = 'To create a RP, a resource group is required. Please add ' \
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/backup/tests/latest/preparers.py:284]:
```suggestion
            template = 'To create an RP, a resource group is required. Please add ' \
```

File: src/azure-cli/requirements.py3.windows.txt (hunk 1 above)
Comment (qwordy) [reply src/azure-cli/requirements.py3.windows.txt]: There are also occurrences of this issue in this PR. Please fix them, too.

**Comment by qwordy** [comment]
Please fix "a RP".

**Comment by PARADISSEEKR** [comment]
> Please fix "a RP".

It has been fixed

**Comment by qwordy** [comment]
@andyliuliming, @fengzhou-msft, @zqingqing1 Could you help review this PR?

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_acs_create_kubernetes.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+interactions:
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_acs_create_kubernetes.yaml:1]: This file should not be included in this PR.
